site2/website/contributing.md

```diff
@@ -0,0 +1,250 @@
+
+We require you to have an ICLA on file with the Apache Secretary for larger contributions only. For smaller ones, however, we rely on [clause five](http://www.apache.org/licenses/LICENSE-2.0#contributions) of the Apache License, Version 2.0, describing licensing of intentionally submitted contributions and do not require an ICLA in that case.
+
+#### Obtain a GitHub account
+
+We use GitHub’s pull request functionality to review proposed code changes.
+
+If you do not already have a personal GitHub account, sign up [here](https://github.com/join).
+
+#### Fork the repository on GitHub
+
+Go to the [Pulsar GitHub Repo](https://github.com/apache/incubator-pulsar/) and fork the repository
+to your own private account. This will be your private workspace for staging changes.
+
+#### Clone the repository locally
+
+You are now ready to create the development environment on your local machine. Feel free to repeat
+these steps on all machines that you want to use for development.
+
+We assume you are using SSH-based authentication with GitHub. If necessary, exchange SSH keys with
+GitHub by following [their instructions](https://help.github.com/articles/generating-an-ssh-key/).
+
+Clone your personal Pulsar’s GitHub fork.
+
+    $ git clone https://github.com/<Github_user>/incubator-pulsar.git
+    $ cd incubator-pulsar
+
+Add Apache Repo as additional Git remotes, where you can sync the changes (for committers, you need
+these two remotes for pushing changes).
+
+	$ git remote add apache https://github.com/apache/incubator-pulsar
+
+You are now ready to start developing!
+
+#### [Optional] IDE Setup
+
+Depending on your preferred development environment, you may need to prepare it to develop Pulsar code.
+
+##### IntelliJ
+
+###### Enable Annotation Processing
+
+To configure annotation processing in IntelliJ:
+
+1. Open Annotation Processors Settings dialog box by going to Settings -> Build, Execution, Deployment -> Compiler -> Annotation Processors.
+1. Select the following buttons:
+    1. "Enable annotation processing"
+    1. "Obtain processors from project classpath"
+    1. "Store generated sources relative to: Module content root"
+1. Set the generated source directories to be equal to the Maven directories:
+    1. Set "Production sources directory:" to "target/generated-sources/annotations".
+    1. Set "Test sources directory:" to "target/generated-test-sources/test-annotations".
+1. Click "OK".
+
+##### Eclipse
+
+Use a recent Eclipse version that includes m2e. Currently we recommend Eclipse Neon.
```

Comment:
> Currently we recommend Eclipse Neon.

Are you sure? That's an ancient version.

Joking aside, they release a new version every year so it gets obsolete soon. I don't think we need to recommend a specific version. It'd makes more sense if we wrote a version in the Initial setup section because the steps can be changed.

Reply:
Fixed